up-l2/dispatchers/README.adoc

```diff
@@ -30,95 +30,88 @@ SPDX-License-Identifier: Apache-2.0
+  ** *MUST* queue messages not successfully acknowledged (transport level at-least-once delivery confirmation described above)
+  ** *MUST* attempt to retry transmission of the message. Retry policy is specific to the dispatcher implementation
+  ** Dispatcher *MUST NOT* discard message unless either the message has expired (see https://github.com/eclipse-uprotocol/uprotocol-core-api/blob/main/src/main/proto/uattributes.proto[`UAttribute:ttl`] for more details), or the egress queue is full. Messages that cannot be delivered are sent to a Dead Letter Office Topic (DLT)
+
+NOTE: Implementation of the DLT is uPlatform specific and is not covered in this document 
 
-* *MAY* support additional CE delivery policies in general or per topic in the future
+* *MAY* support additional message delivery policies in general or per topic in the future
 * *SHOULD* dispatch in order that it received the CE
 * *MAY* batch CEs when delivering to the Receiver
 * CEs that cannot be delivered *MUST* be sent to the Dead Letter topic (DLT)
-  ** DLT *MUST* include at least the CE header, SHOULD contain the full CE
-  ** DLT *MUST* include the reason for the failed delivery attempt using  error codes defined in google.rpc.Code
-  ** uEs MUST be able to subscribe to the DLT to be notified of message deliver failures
+  ** DLT *MUST* include at least the message header, *SHOULD* contain the full Message
+  ** DLT *MUST* include the reason for the failed delivery attempt using per https://github.com/eclipse-uprotocol/uprotocol-core-api/blob/main/src/main/proto/ustatus.proto[UStatus] error codes
+  ** uEs *SHOULD* be able to subscribe to the DLT to be notified of message deliver failures
 
 If the uP-L1 delivery method is push:
 
-* *SHALL* provide an API to start/stop dispatching of CEs per-topic, this is to avoid having to queue CEs on the Receiver if the Receiver is not ready to receive the CEs
+* *SHOULD* provide an API to start/stop dispatching of messages per-topic, this is to avoid having to queue messages on the Receiver if the Receiver is not ready to receive
 
 
 == RPC Error Handling
 
-When a dispatcher is unable to dispatch an event for a given reason (queue full, etc...), the dispatcher is responsible to generate an RPC Response message and send it to the originator of the request. The status code is populated in the  link:../messages/v1/README.adoc#_response_message[`commstatus`] attribute of the response message using the <<commstatus-codes>> defined below.
+When a dispatcher is unable to dispatch an event for a given reason (e.g. queue full, etc...), the dispatcher is responsible to generate an RPC Response message and send it to the originator of the request. The status code is populated in the  link:../messages/v1/README.adoc#_response_message[`commstatus`] attribute of the response message using the error codes defined in https://github.com/eclipse-uprotocol/uprotocol-core-api/blob/main/src/main/proto/ustatus.proto[UCode].
```

Comment:
so, if the dispatcher could not deliver a message it sends another RPC that can't be delivered for the same reason the first event could not be delivered hence adding more and more messages that can't be delivered, bogging down the system.
IMHO dispatchers so their best to send the messages, using brokers, using caches using whatever. Either the subscribed uE gets its data or it does not. Why do we need another message saying that things failed? How does that help?
I think I am not understanding this correctly. Can we please clarify?